kratos/sources/model_part.cpp

```diff
@@ -400,38 +400,61 @@ void ModelPart::RemoveNodeFromAllLevels(ModelPart::NodeType::Pointer pThisNode,
 
 void ModelPart::RemoveNodes(Flags IdentifierFlag)
 {
-    // This method is optimized to free the memory
-    //loop over all the meshes
-    ModelPart::MeshesContainerType& meshes = this->GetMeshes();
-    for(ModelPart::MeshesContainerType::iterator i_mesh = meshes.begin() ; i_mesh != meshes.end() ; i_mesh++)
-    {
+    // Lambda to remove nodes from a mesh
+    auto remove_from_mesh = [&](ModelPart::MeshType & mesh) {
```

Comment:
```suggestion
    auto remove_nodes_from_mesh = [&](ModelPart::MeshType& r_mesh) {
```

(at least to me) this is much more readable/understandable